Source/WebCore/platform/mediastream/gstreamer/GStreamerVideoEncoder.cpp

```diff
@@ -460,6 +462,23 @@ static void webkit_webrtc_video_encoder_class_init(WebKitWebrtcVideoEncoderClass
             g_object_set(self->priv->inputCapsFilter.get(), "caps", inputCaps.get(), nullptr);
         }, "target-bitrate", setBitrateBitPerSec, "keyframe-max-dist");
 
+    Encoders::registerEncoder(VaapiH264, "vah264lpenc", "h264parse", "video/x-h264", nullptr,
+        [](WebKitWebrtcVideoEncoder* self) {
+            g_object_set(self->priv->parser.get(), "config-interval", 1, nullptr);
+        }, "bitrate", setBitrateKbitPerSec, "key-int-max");
+
+    Encoders::registerEncoder(VaapiH264, "vah264enc", "h264parse", "video/x-h264", nullptr,
+        [](WebKitWebrtcVideoEncoder* self) {
+            gst_util_set_object_arg(G_OBJECT(self->priv->encoder.get()), "rate-control", "cbr");
+            g_object_set(self->priv->parser.get(), "config-interval", 1, nullptr);
+        }, "bitrate", setBitrateKbitPerSec, "key-int-max");
+
+    Encoders::registerEncoder(VaapiH265, "vah265enc", "h265parse", "video/x-h265", nullptr,
+        [](WebKitWebrtcVideoEncoder* self) {
+            gst_util_set_object_arg(G_OBJECT(self->priv->encoder.get()), "rate-control", "cbr");
+            g_object_set(self->priv->parser.get(), "config-interval", 1, nullptr);
+        }, "bitrate", setBitrateKbitPerSec, "key-int-max");
+
```

Comment:
Let's keep in mind that vah265enc was merged in 1.22 in extremis with little review :) So it's quite experimental right now. We want to use these release to enhance it as much as possible.

Reply:
Right. It's rank reflects that too, which means by default it's not going to be used anyway :) All this VA encoder support is opt-in.